src/components/Map/ChallengeMap.tsx

```diff
@@ -34,6 +37,29 @@ const ChallengeMap = ({ tracks, challenge, setChallenge }: PropTypes) => {
     },
     [],
   )
+  React.useEffect(() => {
```

Comment:
**Notes about ViewState**
The inner-workings of the VS are going over my head right now. Piecing it apart, I can identify most of the differing components, but I cannot tell as a whole whether there are any hidden bugs right now.

Long story short, it looks good to me :)

Reply:
Ya it can be a little picky with what it wants so luckily the parts we need make our functions work the way they are supposed to and if anything breaks along the way we can always fix it 👍🏼